plugins/module_utils/network/sonic/sonic.py

```diff
@@ -139,6 +139,24 @@ def edit_config(module, commands, skip_code=None):
     return connection.edit_config(commands)
 
 
+def edit_config_catch(module, commands, skip_code=None):
+    connection = get_connection(module)
+
+    # Start: This is to convert interface name from Eth1/1 to Eth1%2f1
+    for request in commands:
+        # This check is to differenciate between requests and commands
+        if isinstance(request, dict):
+            url = request.get("path", None)
+            if url:
+                request["path"] = update_url(url)
+    # End
+    try:
+        response = connection.edit_config(commands)
+    except ConnectionError:
+        response = [[{}, {}]]
+    return response
+
+
```

Comment:
I Suppose this new function will be used by "GET" request only. It is proper to return "[[{}, {}]]" if target system return error 404 (resource not found). How about other errors returned from target system? or not expect other errors from target system for "GET" request?

Reply:
Yes. I think this handling should be conditional on a 404 error, maybe only when the request is a GET. I am also thinking that the fix needs to be lower in the call stack to enable handling for an individual request in the list of requests. (It is currently handing an exception occurring anywhere during the iterative loop through the requests at the "connection type" edit_config function.)

I will do some rework on the fix to address these concerns.

Reply:
I have modified the change set to use the suggested filtering criteria plus an additional flag specified by the application to enable/disable the filtering.